geoapps/inversion/components/topography.py

```diff
@@ -96,7 +97,23 @@ def active_cells(self, mesh: InversionMesh, data: InversionData) -> np.ndarray:
             print(
                 "Adjusting active cells so that receivers are all within an active cell . . ."
             )
-            active_cells[get_containing_cells(mesh.mesh, data)] = True
+            containing_cells = get_containing_cells(mesh.mesh, data)
```

Comment:
This function is starting to get too complicated.  Can we pull out your additions to a new function?  Also, some common code we can pull out:
```
grid_reference = "bottom" if self.params.inversion_type in forced_to_surface else center"
active_cells = active_from_xyz(
    mesh.entity, self.locations, grid_reference=grid_reference
)
active_cells = active_cells[np.argsort(mesh.permutation)]
if self.params.inversion_type in forced_to_surface:
    ...
    if isinstance(mesh.mesh, TreeMesh):
        self._ensure_tree_mesh_connectivity()
```

Reply:
...then drop the bottom else statement

Reply:
Good idea. Done.